gflow/gflow.py

```diff
@@ -1,107 +1,255 @@
+        try:
+            self.galaxy_url = datadict['galaxy_url']
+            self.galaxy_key = datadict['galaxy_key']
+            self.library_name = datadict['library_name']
+            self.history_name = datadict['history_name']
+            self.workflow_source = datadict['workflow_source']
+            self.workflow = datadict['workflow']
+        except KeyError as e:
+            self.logger.error("Missing required parameter: " + str(e))
+            raise KeyError
+        # Optional parameters follow
+        self.datasets_source = None
+        self.datasets = None
+        self.runtime_params = None
+        try:
+            self.datasets_source = datadict['datasets_source']
+            self.datasets = datadict['datasets']
+            self.runtime_params = datadict['runtime_params']
+        except KeyError as e:
+            self.logger.warning("%s parameter not set", e)
+
+    @classmethod
+    def from_config_file(cls, configfile):
+        """
+        Makes a GFlow object from a config file
+
+        Args:
+            configfile (str): The name of the config file to be read
+        """
+        cls.logger = logging.getLogger('gflow.GFlow')
+        cls.logger.info("Reading configuration file")
+        with open(configfile, "r") as ymlfile:
+            config = yaml.load(ymlfile)
+        GFlow.check_for_empty_values(cls, config)
+        return cls(config)
+
+    @classmethod
+    def from_params(cls, galaxy_url=None, galaxy_key=None, library_name=None, history_name=None,
+                    workflow_source=None, workflow=None, datasets_source=None, datasets=None, runtime_params=None):
+        """
+        Makes GFlow object from provided parameters
+
+        Args:
+            galaxy_url (str): The URL of an instance of Galaxy
+            galaxy_key (str): The API key of an instance of Galaxy
+            library_name (str): The name of the library to be created
+            history_name (str): The name of the history to be created
+            workflow_source (str): Whether the workflow's being imported from a file or with an id
+            workflow (str): Either a filename or id for the workflow
+            datasets_source (str): Whether the data's being imported from files or URLs
+            datasets (List): A collection of filenames or URLs for the datasets
+            runtime_params (dict): A collection of required runtime parameters
+        """
+        cls.logger = logging.getLogger('gflow.GFlow')
+        cls.logger.info("Reading from parameters")
+        config = {'galaxy_url': galaxy_url, 'galaxy_key': galaxy_key, 'library_name': library_name,
+                  'history_name': history_name, 'workflow_source': workflow_source, 'workflow': workflow,
+                  'datasets_source': datasets_source, 'datasets': datasets, 'runtime_params': runtime_params}
+        return cls(config)
+
+    @staticmethod
+    def check_for_empty_values(cls, value):
```

Comment:
Return true for successful execution and check for it where you use the method. It is generally a good idea not to rely on throwing exception as part of your normal execution flow.